### PR TITLE
Make Reindex cancellation tests more uniform

### PR DESCRIPTION
**Pull request by tlrx**
This pull request makes all the Reindex cancellation tests more uniform. They now all inherit from the same abstract class `ReindexCancelTestCase` and follow the same behaviour.

It also fixes a small bug where the `Task` status was not updated if the task was cancelled even if the operations were proceeded.

This relates to https://github.com/elastic/elasticsearch/pull/18329#discussion_r63392207

### CODE REVIEW

**Comment by tlrx** [comment]
@nik9000 Can you have a look? Thanks

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryCancelTests.java [hunk 1]
```diff
@@ -19,166 +19,24 @@
...
- * not exercise important portion of the stack like transport and task management.
- */
-public class DeleteByQueryCancelTests extends ReindexTestCase {
 
-    private static final String INDEX = "test-delete-by-query";
-    private static final String TYPE = "test";
-
-    private static final int MAX_DELETIONS = 10;
-    private static final CyclicBarrier barrier = new CyclicBarrier(2);
+public class DeleteByQueryCancelTests extends ReindexCancelTestCase<DeleteByQueryRequestBuilder> {
```
Comment (nik9000) [inline modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryCancelTests.java]: I'd just make one CancelTests or something and have it check all three actions.

**Comment by nik9000** [comment]
I left a one comment about squashing the tests into a single file but otherwise LGTM.

**Comment by tlrx** [comment]
Thanks @nik9000 ! I merged all 3 tests in a `CancelTests` class.

**Pull request merged**